src/lib/fieldValidators.ts

```diff
@@ -0,0 +1,86 @@
+export interface IFieldValidator {
+  validate(value: string): IFieldValidationResponse;
+}
+
+export interface IFieldValidationResponse {
+  value: string;
+  errors: string[];
+  valid: boolean;
+  invalid: boolean;
```

Comment:
I found having both a `valid` and `invalid` method was useful for readability on the page.  E.g.:

```
<ExampleComponent hasErrors={ExampleFieldValidator.invalid} />
<AnotherExampleComponent isValid={ExampleFieldValidator.valid} />
```